src/GroupPermissions.php

```diff
@@ -44,15 +47,19 @@ public function initPermissions( &$vars ) {
 
 			// Rights
 			foreach ( array_keys( $rights ) as $right ) {
-				$vars['wgAvailableRights'][] = $right;
+				$newVars['wgAvailableRights'][] = $right;
 			}
 
 			if ( !isset( $vars['wgGroupPermissions'][$group] ) ) {
-				$vars['wgGroupPermissions'][$group] = [];
+				$newVars['wgGroupPermissions'][$group] = [];
 			}
 
-			$vars['wgGroupPermissions'][$group] = array_merge( $rights, $vars['wgGroupPermissions'][$group] );
+			$newVars['wgGroupPermissions'][$group] = array_merge( $rights, $vars['wgGroupPermissions'][$group] );
 		}
+
+		Globals::replace( $newVars );
```

Comment:
Can't the caller do this side effect? Weird to both return the new value and modify globals. Like this the test also does not check the production behavior.